spring-xd-module/src/main/java/org/springframework/xd/module/ModuleDefinition.java

```diff
@@ -76,21 +73,37 @@ public int compareTo(ModuleDefinition other) {
 	}
 
 	@Override
-	public final boolean equals(Object o) {
-		if (this == o) return true;
-
-		ModuleDefinition that = (ModuleDefinition) o;
-
-		if (!name.equals(that.name)) return false;
-		if (type != that.type) return false;
-
+	public boolean equals(Object obj) {
+		if (this == obj) {
+			return true;
+		}
+		if (obj == null) {
+			return false;
+		}
+		if (getClass() != obj.getClass()) {
+			return false;
+		}
+		ModuleDefinition other = (ModuleDefinition) obj;
+		if (name == null) {
```

Comment:
OK; I simply used eclipse to generate; Sonar complained that `equals()` should never assume the passed-in object is a `ModuleDefinition` (previously the cast was done unconditionally).

If someone calls `myModDef.equals("foo")` we'd get a `ClassCastException`.

Maybe we should revert and add a check for the type; but using `equals()` in this way seems like a bastardization to me.